react-validated-input-component/my-app/src/ValidatedInput.js

```diff
@@ -37,25 +35,24 @@ export default function ValidatedInput() {
       const upper = /[A-Z]/;
       const noUpper = 'Password must include at least one uppercase letter.';
       !upper.test(typedPw) ? message.push(noUpper) : message.filter((msg) => msg !== noUpper);
-      setErr(message.map((errorMessage) => <p key={errorMessage} style={msgStyle}>{errorMessage}</p>));
+      return setErr(message);
     }
+    return setErr(message);
```

Comment:
It doesn't look to me like this still accumulates the error messages and displays them all. You return from this after finding just one error. Is that what you now want to do? It doesn't look like it, since you are trying to `setErr` to an array of messages. (Recall that since `message` is a local variable, it gets reset to `[]` on every render.)

The best way to update a state variable based on its existing value (such as adding an element to the list) is to use the function version:
`setErr((prev) => [...prev, msg])`
Keep in mind that when you use this, you must return a _new_ array. If you simply mutate the array (by using `push`, for example), then it will _not_ trigger a re-render.